### PR TITLE
[GPU] AdaptiveMaxPool and AdaptiveAvgPool gpu implementations

### PR DESCRIPTION
**Pull request by tgubanova-lohika**
- Add kernel for AdaptivePooling
- Add GPU primitive for AdaptivePooling
- Add single-layer tests for GPU

### CODE REVIEW

**Comment by wilson-seok** [comment]
@tgubanova-lohika We need clDNN unit test for adaptive pooling primitives.

**Comment by tgubanova-lohika** [comment]
> @tgubanova-lohika We need clDNN unit test for adaptive pooling primitives.

done

File: src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/adaptive_pooling.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using namespace ngraph::helpers;
+using namespace LayerTestsDefinitions;
+using namespace ngraph::element;
+
+namespace {
+const std::vector<std::string> poolingModes = {"max", "avg"};
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::I32,
```
[the remark this answers is not in the document]
Comment (tgubanova-lohika) [reply src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/adaptive_pooling.cpp]: done

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/adaptive_pooling/adaptive_pooling_kernel_ref.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#include "kernel_selector_utils.h"
+#include <algorithm>
+#include <string>
+
+namespace kernel_selector {
+
+ParamsKey AdaptivePoolingRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
```
Comment (wilson-seok) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/adaptive_pooling/adaptive_pooling_kernel_ref.cpp:15]: Int data type is actually supported in kernel? From the spec of the op, only floating point is supported. It the data type is supported, it should be tested in unit test/single layter test.

File: src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/adaptive_pooling.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wilson-seok) [reply src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/adaptive_pooling.cpp]: Thanks!

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/adaptive_pooling/adaptive_pooling_kernel_ref.cpp (hunk 2 above)
Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/adaptive_pooling/adaptive_pooling_kernel_ref.cpp:15]: you are right the first input is float always, but we also have second indices input whis is Int. That's why we need to support it as input type also, right?

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/adaptive_pooling/adaptive_pooling_kernel_ref.cpp:15]: Got it! Thanks!

**Review by wilson-seok — APPROVED** [review]
LGTM

**Comment by wilson-seok** [comment]
Please rebase the PR to resolve issue in openvino-mac check.

File: src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            std::make_tuple(data_types::f16, format::bs_fs_zyx_bsv32_fsv16),
+            std::make_tuple(data_types::f16, format::b_fs_yx_fsv32),
+            std::make_tuple(data_types::f32, format::bs_fs_yx_bsv32_fsv32),
+
+            std::make_tuple(data_types::f32, format::bfyx),
+            std::make_tuple(data_types::f32, format::bfzyx),
+            std::make_tuple(data_types::f32, format::b_fs_yx_fsv16),
+            std::make_tuple(data_types::f32, format::bs_fs_yx_bsv16_fsv16),
+            std::make_tuple(data_types::f32, format::b_fs_zyx_fsv16),
+            std::make_tuple(data_types::f32, format::bs_fs_zyx_bsv32_fsv16),
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp:93]: please add this operation to the operation whitelist in program.cpp for can_use_fsv16, can_use_bsv32_fsv...

File: src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                         "Cannot find a proper kernel with this arguments");
+
+        return new adaptive_pooling_impl(arg, best_kernels[0]);
+    }
+};
+
+namespace detail {
+attach_adaptive_pooling_impl::attach_adaptive_pooling_impl() {
+    implementation_map<adaptive_pooling>::add(impl_types::ocl, adaptive_pooling_impl::create, {
+            std::make_tuple(data_types::f16, format::bfyx),
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp:79]: what about i8 and  u8?

File: src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp (hunk 3 above)
Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/src/graph/impls/ocl/adaptive_pooling.cpp]: As we discussed today, removed these layouts for now

**Pull request merged**